### PR TITLE
Use safe_constantize to mitigate attack vector for dynamic badge slug

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/jobs/award_badge_job.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def self.worth_queuing?(badge_slug, context:)
     return true if context.blank?
 
     context_key = context.class.base_class.name.demodulize.underscore
-    badge = "Badges::#{badge_slug.to_s.camelize}Badge".constantize
+    badge = "Badges::#{badge_slug.to_s.camelize}Badge".safe_constantize
```
Comment (ErikSchierboom) [inline app/jobs/award_badge_job.rb:20]: @SleeplessByte I've opted for the `safe_constantize` option, as the `badge.worth_queuing?` call on the next line will then fail with a nil reference error, which we'll automatically log.

**Review by iHiD — APPROVED** [review]

**Comment by iHiD** [comment]
So I'm clear, do we have an example of what this protects us against?

**Comment by ErikSchierboom** [comment]
> So I'm clear, do we have an example of what this protects us against? 

@SleeplessByte ?

**Comment by SleeplessByte** [comment]
I'd need to see if I can craft an attack. I have no proof at the moment that it's possible, but whatever you do, whenever you call `constantize` it is important that you whitelist (which this PR does not do). A next refactor may remove `Badge::` from the string, and then it becomes "dangerous".

**Comment by ErikSchierboom** [comment]
It sounds like it requires us to merge code before there is an attack vector? Is that correct?

**Comment by SleeplessByte** [comment]
I think so, yes. Which is good!

**Pull request merged**